### PR TITLE
{Homebrew} Hotfix: Build with template to revert to python 3.8

### PR DESCRIPTION
**Pull request by fengzhou-msft**
**Description**  

https://github.com/Homebrew/homebrew-core/pull/62253 migrated to Python 3.9 for Azure CLI, but Azure CLI is not compatible with Python 3.9 yet. ADO has not supported Python 3.9 either and it caused failures when testing homebrew formula, which blocked the release of our packages.

This PR updates the homebrew template to sync the changes with Homebrew master branch except that Python is still 3.8 and build the homebrew formula from the template.

**Testing Guide**  

Tested the changes in `build_test` branch: https://dev.azure.com/azure-sdk/public/_build/results?buildId=570383&view=results

**History Notes**  

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.  
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Review by arrownj — COMMENTED** [review]
Actually I still doubt that whether homebrew will update the python version even we pin python version in the formula.

File: scripts/release/homebrew/docker/formula_template.txt [hunk 1]
```diff
@@ -9,6 +9,11 @@ class AzureCli < Formula
...
   head "https://github.com/Azure/azure-cli.git"
 
+  livecheck do
+    url "https://github.com/Azure/azure-cli/releases/latest"
+    regex(%r{href=.*?/tag/azure-cli[._-]v?(\d+(?:\.\d+)+)["' >]}i)
+  end
+
 {{ bottle_hash }}
 
   depends_on "openssl@1.1"
```
Comment (fengzhou-msft) [inline scripts/release/homebrew/docker/formula_template.txt:19]:
```suggestion
  depends_on "openssl@1.1"
  # Azure CLI is not compatible with Python 3.9 yet
```

File: scripts/release/homebrew/docker/formula_template.txt [hunk 2]
```diff
@@ -9,9 +9,15 @@ class AzureCli < Formula
   license "MIT"
   head "https://github.com/Azure/azure-cli.git"
 
+  livecheck do
+    url "https://github.com/Azure/azure-cli/releases/latest"
+    regex(%r{href=.*?/tag/azure-cli[._-]v?(\d+(?:\.\d+)+)["' >]}i)
```
Comment (jiasli) [inline scripts/release/homebrew/docker/formula_template.txt:14]: Can we add some comments about what this regex does?

**Comment by fengzhou-msft** [comment]
> Actually I still doubt that whether homebrew will update the python version even we ping python version in the formula.

Valid concern, based on their migration plan: https://github.com/Homebrew/homebrew-core/issues/62201, they only test whether the formula can be installed locally. I added a comment to let them know Azure CLI is not compatible with Python 3.9.

File: scripts/release/homebrew/docker/formula_template.txt (hunk 2 above)
Comment (fengzhou-msft) [reply scripts/release/homebrew/docker/formula_template.txt:14]: It's added by Homebrew team, seems to parsing the version number from `https://github.com/Azure/azure-cli/releases/latest` and then do [livecheck](https://github.com/Homebrew/homebrew-livecheck).

**Review by haroldrandom — APPROVED** [review]

**Review by jiasli — APPROVED** [review]

**Review by yungezz — APPROVED** [review]
hi @fengzhou-msft could you pls help to follow up how to get change nofitication on homebrew, we met surprise twice in past releases. It will be helpful if we could get earlier notification.

**Comment by fengzhou-msft** [comment]
> hi @fengzhou-msft could you pls help to follow up how to get change nofitication on homebrew, we met surprise twice in past releases. It will be helpful if we could get earlier notification.

The change is made through PRs as an open source project. They don't have CODEOWNERS in their repository.
There are some third-party tools to get notified when some file changes on Github such as https://app.github-file-watcher.com/

**Comment by yungezz** [comment]
> > hi @fengzhou-msft could you pls help to follow up how to get change nofitication on homebrew, we met surprise twice in past releases. It will be helpful if we could get earlier notification.
> 
> The change is made through PRs as an open source project. They don't have CODEOWNERS in their repository, which is an ideal way to get notified for PRs.
> There are some third-party tools to get notified when some file changes on Github such as https://app.github-file-watcher.com/

:( let me comment at the our previous PR to feedback about this.

**Pull request merged**